Display tool source in tool form

mvdbeek opened the pull request:
<img width="276" alt="Screenshot 2021-10-29 at 19 36 40" src="https://user-images.githubusercontent.com/6804901/139478681-92e744d5-7a0e-42da-987d-f48ee10c0449.png">

<img width="1648" alt="Screenshot 2021-10-29 at 19 36 22" src="https://user-images.githubusercontent.com/6804901/139478665-21ce50a2-6252-42af-91bc-821aa38cf273.png">


TODO: 
- [x] create a standalone component
- [x] return tool source type in header
- [x] detect language
- [x] pass in tool version
- [x] tests
- [x] run some kind of prettify library over raw response
 
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

mvdbeek committed:
Display tool source in tool form

guerler commented on client/src/components/Tool/ToolCard.vue:
```diff
@@ -26,6 +26,9 @@
                             ><span class="fa fa-download" /><span v-localize>Download</span>
                         </b-dropdown-item>
+                        <b-dropdown-item v-b-modal.tool-source-viewer
+                            ><span class="fa fa-eye" /><span v-localize>Preview tool source</span>
```
How about `View Tool source` or `View Tool XML`  in order to align it with `View Tool ID`?

mvdbeek committed:
Return tool source language in header,

disable view unless admin or allow_tool_source_display is set

mvdbeek committed:
Add API tests for raw_tool_source endpoint

mvdbeek committed:
Fix up tests now that only admins can see tool source

bgruening commented on lib/galaxy/webapps/galaxy/config_schema.yml:
```diff
@@ -2821,6 +2821,17 @@ mapping:
           Details" option in the history. This option also exposes the command line to
           non-administrative users. Administrators can always see dataset paths.
 
+      allow_tool_source_display:
```
Maybe `enable_tool_source_display` to be consistent with other options (enable_tool_recommendations, enable_markdown_export, enable_tool_tags)? Seems that enable is used more?

mvdbeek committed:
Rename allow_tool_source_display to enable_tool_source_display

mvdbeek committed:
Fix multiple root nodes

I was under the impression one could use template instead of div, but
that obviously doesn't work.

dannon reviewed (approved):
@mvdbeek Thanks for the fixup, looks great!
